pkg/extensions/tbtc/tbtc.go

```diff
@@ -36,29 +38,73 @@ func Initialize(ctx context.Context, handle Handle) error {
 		)
 	}
 
+	err = tbtc.monitorProvideRedemptionSignature(
+		ctx,
+		exponentialBackoff,
+		105*time.Minute, // 15 minutes before the 2 hours on-chain timeout
+	)
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize provide redemption "+
+				"signature monitoring: [%v]",
+			err,
+		)
+	}
+
 	logger.Infof("tbtc extension has been initialized")
 
 	return nil
 }
 
-type depositEventHandler func(deposit string)
+type depositEvent interface {
+	getDepositAddress() string
+}
+
+type baseDepositEvent struct {
+	depositAddress string
+}
+
+func (bdd *baseDepositEvent) getDepositAddress() string {
+	return bdd.depositAddress
+}
+
+type depositRedemptionRequestedEvent struct {
+	*baseDepositEvent
+
+	digest      [32]uint8
+	blockNumber uint64
+}
+
+type depositEventHandler func(depositEvent)
```

Comment:
I prefer the previous version: `type depositEventHandler func(depositAddress string)`. The additional indirection makes it harder to understand what's happening and the logic in app-specific actions is already quite complex.

Reply:
So am I but I explained why we have to introduce them [here](https://github.com/keep-network/keep-ecdsa/pull/585#discussion_r510775802)

Reply:
NVM, I've changed the approach so we have the previous version.